guide/additional-info/changes-in-v13.md

```diff
@@ -0,0 +1,627 @@
+
+## Changes and deletions
+
+### Client
+
+#### Client#emojis
+
+The Client Emoji manager is now a `BaseGuildEmojiManager`, providing cache resolution only and removing methods that would fail to create emojis as there was no Guild context.
+
+#### Client#fetchApplication
+
+This method has been removed and replaced with the `Client#application` property
+
+```diff
+- client.fetchApplication().then(application => console.log(application.name))
++ console.log(client.application.name);
+```
+
+#### Client#generateInvite
+
+`Client#generateInvite` no longer supports `PermissionsResolvable` as its argument, requiring `InviteGenerationOptions`.
+
+To provide permissions, use `InviteGenerationOptions#permissions`.
+```diff
+- client.generateInvite([Permissions.FLAGS.SEND_MESSAGES]);
++ client.generateInvite({ permissions: [Permissions.FLAGS.SEND_MESSAGES] })
+```
+
+`InviteGenerationOptions` also supports passing additional scopes, for OAuth or application.commands scopes.
+
+### Client#login
+
+Previously when a token had reached its 1000 login limit for the day, discord.js would treat this as a rate limit and silently wait to login again, but this was not communicated to the user.
+
+This will now instead cause an error to be thrown.
+
+### ClientOptions
+
+#### ClientOptions#fetchAllMembers
+
+The `ClientOptions#fetchAllMembers` option has been removed.
+
+With the introduction of gateway intents, the `fetchAllMembers` Client option would often fail and causes significant delays in ready states or even cause timeout errors. As its purpose is contradictory to Discord's intentions to reduce scraping of user and presence data, it has been removed.
+
+#### ClientOptions#messageEditHistoryMaxSize
+
+The `ClientOptions#messageEditHistoryMaxSize` option has been removed.
+
+To reduce caching, discord.js will no longer store an edit history. You will need to implement this yourself if required.
+
+### ClientUser
+
+#### ClientUser#setActivity
+
+This method no longer returns a Promise.
+
+#### ClientUser#setAFK
+
+This method no longer returns a Promise.
+
+#### ClientUser#setPresence
+
+`PresenceData#activity` is replaced with `PresenceData#activities` which now requires an `Array<ActivitiesOptions>`.
+
+```diff
+- client.user.setPresence({ activity: { name: 'with discord.js' } });
++ client.user.setPresence({ activities: [{ name: 'with discord.js' }] });
+```
+
+This method no longer returns a Promise.
+
+#### ClientUser#setStatus
+
+This method no longer returns a Promise.
+
+### Guild
+
+#### Guild#fetchVanityCode
+
+This method has been removed.
+
+```diff
+- Guild.fetchVanityCode().then(code => console.log(`Vanity URL: https://discord.gg/${res.code}`));
++ Guild.fetchVanityData().then(res => console.log(`Vanity URL: https://discord.gg/${res.code} with ${res.uses} uses`));
+```
+
+#### Guild#member
+
+The helper/shortcut method `Guild#member()` has been removed.
+
+```diff
+- guild.member(user);
++ guild.members.cache.get(user.id)
+```
```

Comment:
Perhaps the `Guild#members#resolve` method should be mentioned here since it does pretty much the same as the old `Guild#member`?

Reply:
The resolve method is really more of an internal thing, `cache#get` is more consistent with the rest of the guide